tests/installation/partitioning_lvm.pm

```diff
@@ -17,23 +17,38 @@ sub run() {
     my $self = shift;
 
     send_key "alt-d";
-    sleep 2;
 
-    send_key "alt-l", 1;    # enable LVM-based proposal
-    if (get_var("ENCRYPT")) {
-        send_key "alt-y", 1;
-        assert_screen "inst-encrypt-password-prompt";
-        type_password;
-        send_key "tab";
-        type_password;
-        send_key "ret", 1;
-        assert_screen "partition-cryptlvm-summary";
+    if (check_screen('inst-partition-radio-buttons', 10)) {    # detect whether new (Radio Buttons) YaST behaviour
+        if (get_var("ENCRYPT")) {
+            send_key 'alt-e';
+            assert_screen 'inst-encrypt-password-prompt';
+            type_password;
+            send_key 'tab';
+            type_password;
+            send_key 'ret';
+        }
+        else {
+            send_key 'alt-l';
+        }
+        send_key 'alt-o';
```

Comment:
Yes, because the old style of needling is wrong - we want to assert the
summary page in the partitioner frame, that the YaST partitioner proposal
is correct

We don't want to assert that the checkboxes were ticked right (which is
what the old behaviour is doing, incorrectly IMHO..that's not showing the
summary)

On 8 March 2016 at 16:04, Jozef Pupava <notifications@github.com> wrote:

> In tests/installation/partitioning_lvm.pm
> <https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/1139#discussion_r55370684>
> :
>
> > -        type_password;
> > -        send_key "ret", 1;
> > -        assert_screen "partition-cryptlvm-summary";
> > +    if (check_screen('inst-partition-radio-buttons', 10)) {    # detect whether new (Radio Buttons) YaST behaviour
> > +        if (get_var("ENCRYPT")) {
> > +            send_key 'alt-e';
> > +            assert_screen 'inst-encrypt-password-prompt';
> > +            type_password;
> > +            send_key 'tab';
> > +            type_password;
> > +            send_key 'ret';
> > +        }
> > +        else {
> > +            send_key 'alt-l';
> > +        }
> > +        send_key 'alt-o';
>
> This will close disk proposal before assert summary
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/1139/files#r55370684>
> .
>